### PR TITLE
WiFi-1827. Updates for RRM Channel Hop based on Noise Floor threshold

### PR DESCRIPTION
**Pull request by netexp-ravi**
Average Noise Floor is calcluated for the last few samples based on the
configured Noise Floor time. This calculated avg Noise Floor is compared
against the configured Noise Floor threshold to decide if a Channel hop
is required.

Signed-off-by: ravi vaishnav <ravi.vaishnav@netexperience.com>

### CODE REVIEW

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_ovsdb.c [hunk 1]
```diff
@@ -415,6 +423,11 @@ void rrm_update_rrm_config_cb(ovsdb_update_monitor_t *self)
 		rrm_config->schema.snr_percentage_drop = 0;
 		rrm_config->schema.client_disconnect_threshold = 0;
 		rrm_config->schema.probe_resp_threshold = 0;
+		rrm_config->schema.noise_floor_thresh = 0;
+		rrm_config->schema.noise_floor_time = 0;
+		rrm_config->schema.non_wifi_thresh = 0;
+		rrm_config->schema.non_wifi_time = 0;
+		rrm_config->schema.obss_hop_mode = 0;
```
Comment (RickSommerville) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_ovsdb.c:431]: Should we collapse this whole thong to a memset 0 on the structure?

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c [hunk 2]
```diff
@@ -104,6 +104,53 @@ void get_channel_bandwidth(const char* htmode, int *channel_bandwidth)
...
+ * configuration time and the sample time.
+ * Example: If the config_time is 120 sec, and sample time is 15 sec, then number of
+ * samples required for averaging is 120/15 = 8 samples.
+ */
+int rrm_calculate_avg_noise_floor(rrm_entry_t *rrm_data, int nf, int config_time, int sample_time)
+{
+	int ii, avg = 0;
+	int num_samples = config_time/sample_time;
+
+	if (num_samples >= RRM_MAX_NF_SAMPLES)
```
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c:123]: Agreed. I will make that change.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c [hunk 3]
```diff
@@ -104,6 +104,53 @@ void get_channel_bandwidth(const char* htmode, int *channel_bandwidth)
...
+	{
+		rrm_data->rrm_chan_nf_num_el++;
+		return 0;
+	}
+
+	/* Calculate the average */
+	for (ii = 0; ii < num_samples; ii++)
+	{
+		avg += rrm_data->rrm_chan_nf_samples[ii];
+	}
```
[the remark this answers is not in the document]
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c:142]: Agreed. 
Averaging the logarithmic values is not right. I will try and convert dBm to mW, average them, and re-convert to dBm.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_ovsdb.c (hunk 1 above)
Comment (netexp-ravi) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_ovsdb.c:431]: Looks to be safe to do a memset 0. We are anyways removing/deleting/freeing the whole of rrm_config structure.

**Comment by RickSommerville** [comment]
Closed replaced by PR 269

**Pull request closed without merging**